Helper/ApiData.php

```diff
@@ -192,6 +219,8 @@ public function processApiRequest(
                     'href' => $response["data"]->getHref()
                 ]);
 
+                $this->pendingOrderEmailConfirmation->pendingOrderEmailSend($order);
```

Comment:
Consider the role of this code and it's function. it is making a request to the Paytrail Api, that is all that is expected of this processApiRequest function. Having it suddently send an email message during the making of the payment request feels like it is out of place.

Please consider refactoring your code to have the entry point for it in a place that is appropriate for the role and functionality. 

Paytrail module is a bad example for good coding practices as it contains a lot of legacy code particularly in the helpers. But in this case the modification you're introducing makes the code harder to read and more importantly, to predict and understand without reading it line by line. This increases maintenance costs and makes our work harder in the future.

Reply:
Trying to send email using observer, but can't get correct order with all of the data, cause sales_order doesn't create order yet.
Need to put sending code into redirecting class, before redirect. Code validates that responseData is correct and send confirmation email before redirecting to third-party site.